felix/dataplane/linux/int_dataplane.go

```diff
@@ -2831,7 +2831,7 @@ func startBPFDataplaneComponents(
 	if config.bpfProxyHealthzServer == nil && config.KubeProxyHealtzPort != 0 {
 		healthzAddr := fmt.Sprintf(":%d", config.KubeProxyHealtzPort)
 		config.bpfProxyHealthzServer = k8shealthcheck.NewProxyHealthServer(
-			healthzAddr, config.KubeProxyMinSyncPeriod)
+			healthzAddr, config.KubeProxyMinSyncPeriod, nil)
```

Comment:
Need to update this part similar to the upstream kube proxy implementation here: https://github.com/kubernetes/kubernetes/pull/133059/files#diff-962b78049692dfaf72923ea0eabea310b600f2e29011c837bdb49056e797ab34R210-R246
Will need to spend more time to understand so I'll address it in a separate PR.